s3watcher/lib/Upload.js

```diff
@@ -44,16 +54,12 @@ module.exports = {
         const uploadRequest = request.post(options);
 
         return Rx.Observable.create(function(observer){
+            // check mime type, if not jpeg notify CW, stop!
+
             uploadRequest.on("response", function(response){
-                if(response.statusCode == 202){
-                    observer.onNext(response);
-                } else {
-                    const failMessage =
-                        "Failed upload with code: " + response.statusCode;
-                    observer.onError(failMessage);
-                }
-           });
-           uploadRequest.on("error", observer.onError.bind(observer));
+                observer.onNext(uploadResult(response));
+            });
+            uploadRequest.on("error", observer.onError.bind(observer));
```

Comment:
Couldn't find the definitive answer to this in the docs. After some fiddling in the node console and figuring out how to simulate error responses with `netcat`:

```sh
while true; do { echo -e 'HTTP/1.1 500 NOPE\r\n'; } | nc -l 8080; done
```

It does not throw an error as long as the server returns a HTTP response (regardless of whether it's a 4xx or 5xx).